tests/gateway/test_client.py

```diff
@@ -471,3 +471,22 @@ def test_cllient_query_with_disallowed_param(mixed_gateway):
 
     with pytest.raises(HTTPError, match=".*The parameter 'model' is not permitted.*"):
         gateway_client.query(route=route.name, data=data)
+
+
+@mock.patch(
+    "mlflow.gateway.constants.MLFLOW_GATEWAY_CLIENT_QUERY_TIMEOUT_SECONDS", new_callable=lambda: 1
+)
+@mock.patch("requests.Session.request")
```

Comment:
I'm not a big fan of using `mock.patch` as a decorator because:

1. `@mock.patch` makes it difficut what needs mock. Does `MlflowGatewayClient` need mock? Does `gateway_client.query` need mock? Only the person who writes the code knows.
2. Unnecessary mocking could lead to incorrect test results.

Reply:
```
@mock.patch(...)
def test_foo():
    a()
    b()
    c()
    
    # what needs mock?
```

```
def test_foo():
    a()
    b()
    with mock.patch(...):
        c()
    
    # c needs mock?
```

Reply:
Yep, good idea. I'll switch to this :)